### PR TITLE
Replaced instanceof RegExp

### LINKED ISSUE #63: instanceOf RegExp doesn't work in jest
There is an `instanceof RegExp` check used in `convertRegExpToProxy`. This doesn't work in jest which runs each test in a separate execution context.

Changing this check to `Object.prototype.toString.call(o) !== "[object RegExp]"` should fix the issue.

### PR DESCRIPTION
**Pull request by georgehb**
This caused a problem in jest which runs each test in a separate execution context
Fixes #63

### CODE REVIEW

File: lib/index.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ function convertProxyToRegExp(o: any, depth: number): any {
 
 function convertRegExpToProxy(o: any, depth: number): any {
   if (typeof o !== 'object' || !o || depth > 2) return o
-  if (!(o instanceof RegExp)) {
+  if (Object.prototype.toString.call(o) !== "[object RegExp]") {
```
Comment (patrickhulce) [inline lib/index.ts:34]:
```suggestion
  if (Object.prototype.toString.call(o) !== '[object RegExp]') {
```

**Review by patrickhulce — COMMENTED** [review]
thanks @georgehb, LGTM (though I have a slight preference for duck typing verification instead)!

can you share in what specific circumstances this is required and add a test? We've been using regexes with pptr-testing-library in jest for years without issues 🤔 

AFAICT, in order for this to affect the test, the regex would need to be coming from another v8 isolate which I'm having trouble visualizing what that looks like/why one would do that :)

**Comment by georgehb** [comment]
So I'm slightly confused as well as to where the error comes from. We use jest-puppeteer to set up the test environment which launches a single puppeteer instance that's used across all of the tests.

I know jest does some amount of sandboxing to keep tests from separate files isolated which is what I think is causing the problem but I don't quite understand how it does it.

I've extracted the relevant parts into [this repo](https://github.com/georgehb/jest-puppeteer-example) which demonstrates the problem.

File: lib/index.ts [hunk 2]
```diff
@@ -31,7 +31,7 @@ function convertProxyToRegExp(o: any, depth: number): any {
 
 function convertRegExpToProxy(o: any, depth: number): any {
   if (typeof o !== 'object' || !o || depth > 2) return o
-  if (!(o instanceof RegExp)) {
+  if (Object.prototype.toString.call(o) !== '[object RegExp]') {
```
Comment (patrickhulce) [inline lib/index.ts:34]:
```suggestion
  // Support the cross-environment use case where `instanceof` is not sufficient, see https://github.com/testing-library/pptr-testing-library/pull/64
  if (Object.prototype.toString.call(o) !== '[object RegExp]') {
```

**Review by patrickhulce — APPROVED** [review]
> We use jest-puppeteer to set up the test environment which launches a single puppeteer instance that's used across all of the tests.

Ah, gotcha. Yes, if you extend puppeteer in the shared context and then create the regexp in each isolated context, that would explain this 👍 

don't worry about a test then, but a comment would be great

Thanks! 🎉

**Pull request merged**